adversarial_text/adversarial_losses.py

```diff
@@ -136,10 +138,12 @@ def virtual_adversarial_loss_bidir(logits, embedded, inputs,
   """Virtual adversarial loss for bidirectional models."""
   logits = tf.stop_gradient(logits)
   f_inputs, _ = inputs
-  weights = _end_of_seq_mask(f_inputs.labels)
+  weights = f_inputs.eos_weights
+  assert weights is not None
 
   perturbs = [
-      _mask_by_length(tf.random_normal(shape=tf.shape(emb)), f_inputs.length)
+      _mask_by_length(
```

Comment:
indentation is off and inputs->f_inputs

_mask_by_length(tf.random_normal(shape=tf.shape(emb)), f_inputs.length)